### PR TITLE
Add Jackett preset for Torznab Indexer

### PR DESCRIPTION
**Pull request by Wyall**
copied from radarr, added "categories", not sure about "apiPath" as it's not in any of the radarr presets.

Goal is just convenience for adding Jackett Indexers

### CODE REVIEW

File: src/NzbDrone.Core/Indexers/Torznab/Torznab.cs [hunk 1]
```diff
@@ -42,6 +42,7 @@ public override IEnumerable<ProviderDefinition> DefaultDefinitions
                 yield return GetDefinition("HD4Free.xyz", GetSettings("http://hd4free.xyz"));
                 yield return GetDefinition("AnimeTosho Torrents", GetSettings("https://animetosho.org", apiPath: @"/feed/nabapi", categories: new int[0], animeCategories: new[] { 5070 }));
                 yield return GetDefinition("Nyaa Pantsu", GetSettings("https://nyaa.pantsu.cat", apiPath: @"/feed/torznab", categories: new int[0], animeCategories: new[] { 5070 }));
+                yield return GetDefinition("Jackett", GetSettings("http://localhost:9117/torznab/YOURINDEXER", categories: new int[0], animeCategories: new[] { 5070 }));
```
Comment (markus101) [inline src/NzbDrone.Core/Indexers/Torznab/Torznab.cs]: Probably want categories, not animeCategories for most Torznab indexers.

**Comment by Taloth** [comment]
Wouldn't it be easier to simply give jackett the sonarr api key and have it add, remove and configure the indexers as required?

**Comment by galli-leo** [comment]
@Taloth This is already "tracked" as an issue on the Jackett Github: https://github.com/Jackett/Jackett/issues/1413 But it probably won't come to fruition, as our Indexer add API is not very generic (see my comment on the above issue)

File: src/NzbDrone.Core/Indexers/Torznab/Torznab.cs [hunk 2]
```diff
@@ -42,6 +42,7 @@ public override IEnumerable<ProviderDefinition> DefaultDefinitions
                 yield return GetDefinition("HD4Free.xyz", GetSettings("http://hd4free.xyz"));
                 yield return GetDefinition("AnimeTosho Torrents", GetSettings("https://animetosho.org", apiPath: @"/feed/nabapi", categories: new int[0], animeCategories: new[] { 5070 }));
                 yield return GetDefinition("Nyaa Pantsu", GetSettings("https://nyaa.pantsu.cat", apiPath: @"/feed/torznab", categories: new int[0], animeCategories: new[] { 5070 }));
+                yield return GetDefinition("Jackett", GetSettings("http://localhost:9117/torznab/YOURINDEXER", categories: new int[] { 5000, 5020, 5030, 5040, 5045, 5050, 5080 }, animeCategories: new[0]));
```
Comment (markus101) [inline src/NzbDrone.Core/Indexers/Torznab/Torznab.cs]: Is the aggregate torznab URL similar to this URL? If so should we use that instead, so when someone inevitably doesn't change the URL they get something that works.

Comment (Panderine) [reply src/NzbDrone.Core/Indexers/Torznab/Torznab.cs]: What would this do? Any documentation regarding this setting?

Comment (markus101) [reply src/NzbDrone.Core/Indexers/Torznab/Torznab.cs]: What do you mean? What setting?

Comment (Panderine) [reply src/NzbDrone.Core/Indexers/Torznab/Torznab.cs]: When adding a new torznab indexer, I get presented 2 options "custom" and "jackett". How do I use the Jackett option?. With custom I copy the link from jackett "torznab" button without issues, but what does the "jackett" option do? "http://localhost:9117/torznab/YOURINDEXER" What is "YOURINDEXER"?

Comment (markus101) [reply src/NzbDrone.Core/Indexers/Torznab/Torznab.cs]: `YOURINDEXER` would need to be replaced with the matching portion from the actual Jackett URL.

**Pull request closed without merging**